File: akka-http-core/src/main/scala/akka/http/model/headers/RangeUnit.scala
/**
 * Copyright (C) 2009-2014 Typesafe Inc. <http://www.typesafe.com>
 */

package akka.http.model.headers

import akka.http.util.{ Rendering, ValueRenderable }

sealed trait RangeUnit extends ValueRenderable

object RangeUnit {
  object Bytes extends RangeUnit {
    def render[R <: Rendering](r: R): r.type = r ~~ "bytes"
  }

  final case class Other(name: String) extends RangeUnit {
    def render[R <: Rendering](r: R): r.type = r ~~ name
  }
}

File: akka-http-core/src/main/java/akka/http/model/japi/headers/RangeUnit.java
/**
 * Copyright (C) 2009-2014 Typesafe Inc. <http://www.typesafe.com>
 */

package akka.http.model.japi.headers;

public abstract class RangeUnit {
    public abstract String name();

    public static RangeUnit create(String name) {
        return new akka.http.model.headers.RangeUnits.Other(name);
    }
}
